### ISSUE TITLE
SMF hash doesn't work

### ISSUE THREAD

**Issue by GeigerCounter** [AuthMe/AuthMeReloaded#1095]
### What behaviour is observed:
I set up AuthMe to integrate with the SQL authentication used by SMF. It successfully hooks into that table and is able to read/write it. When attempting to login with the password I registered my SMF account with, login fails. When I use the console to force-change password I am able to login to the server, but then login to SMF fails. The SMF hash may actually be different than AuthMe thinks it is, or else another step is failing along the way. I've tried salting it with the forum user name, with the salt column provided by SMF and without salt, but the comparison fails every time.

### What behaviour is expected:
I expected to be able to login with the password of the forum account on the server. 

### Steps/models to reproduce:
The actions that cause the issue
Install and configure SMF 2.0.11, create another user and grant full access to the _members table of smf's database. Alter the table to add columns for management of Minecraft data. ( Last login, in game coordinates, etc. ) Configure AuthMe as shown in the paste below. Attempt to use SMF's password field ('passwd' by default ) to retrieve and store password for the user in-game. ( So as to streamline registration by having the player register for the forum and server simultaneously with the same password. )

### Environment description
Spigot 1.9.4 standalone running on Ubuntu 14.04 LTS connecting to MySQL on localhost. MySQL version: 5.5.52-mariaDB-1ubuntu0.14.04.1

### AuthMe build number:
Shows up in the latest build.

### Configuration:
http://paste.ubuntu.com/24001087/

**Comment by ljacqu** [comment]
Sample provided by @CaptainMcClellan 

```
SpamBot - spam@spamspamgloriousspam.spam - spamspamspam
YouDonGoofed - ohhh@oh.no - yeaboiii
WelpRipMyForum - me@poorlifechoices.net - Augggh
MURICA - MURICA@americaaaaa.usa - America
```

correspond to rows (shortened)
```
member_name	passwd	email_address	password_salt
SpamBot	7ddf303ba1aa0ff189e9ca2a314b653395fa8e92	spam@spamspamgloriousspam.spam	80c7
YouDonGoofed	1fdcb5c6c614bcae69e680dbb168eeac7a64414e	ohhhh@oh.no	462f
WelpRipMyForum	c193513a8ff5a17659c56d1347441a91214abe50	me@poorlifechoices.net	79ae
MURICA	c2bfd8008867920cf930294a03d6b88239c25c28	MURICA@americaaaaa.usa	12fa
```

**Comment by CaptainMcClellan** [comment]
Has there been any progress on this? Naturally, authentication is a high priority on a server.

**Comment by sgdc3** [comment]
Moving this to 5.4 as we are going to release 5.3 as a minor update. Expect 5.4 very soon.

**Comment by ljacqu** [comment]
> minor

Are you sure? :P https://github.com/AuthMe/AuthMeReloaded/wiki/5.3-Changelog

**Comment by sgdc3** [comment]
"minor" related to the other updates xD

**Comment by ljacqu** [comment]
Hi @GeigerCounter, using the following configuration: https://hastebin.com/eguwusagum.bash our SMF integration looks alright. You have to register from the forums but if that is given, changing passwords in the forum or in Minecraft works fine for me, as does logging in on both sides (forum and MC). I propose to open another issue for the registration issue - it would be nice to be able to register from MC as well.

I want another issue for it because we should analyze first and have a look at other forums so we can come up with a generic solution.

**Comment by GeigerCounter** [comment]
@ljacqu I'll try with the latest comment and close the issue if this resolves it. As per opening a new issue, could you handle that please? I'm not even sure what exactly it is you're wanting to put down as an issue.

**Comment by ljacqu** [comment]
That sounds like a plan @GeigerCounter , thank you~

**Comment by GeigerCounter** [comment]
@ljacqu I'm having a bit of a brain slog today, so I might not get this done until tomorrow. Could you tell me what column types I need for the SQL database? Also, AuthMe only tries to create the table if it can't find the appropriate columns, right? I already have a table in place, from a fresh SMF install with only two users. I need AuthMe to not shutdown and use the table that's already there.

**Comment by ljacqu** [comment]
@GeigerCounter No problem, please have a look at my config.yml in https://github.com/AuthMe/AuthMeReloaded/issues/1095#issuecomment-335036850 

> I need AuthMe to not shutdown

Not sure what you mean there—if you already use MySQL you can adapt settings and do /authme reload; otherwise you'll need to restart (cannot change database type on reload)

**Comment by ljacqu** [comment]
One small caveat at my config, I'm not sure about the ip columns (ip vs. ip2). Exercise left for the reader ;)

**Comment by GeigerCounter** [comment]
@ljacqu Fixed as of version 5.4 - at least as far as register from forum goes. Did not try register from server as I don't want/need that. As far as I'm concerned, this issue is closed. Feel free to open it again if there's more you want to mess with here.

**Comment by ljacqu** [comment]
Thanks! I'll create a follow-up issue as described in https://github.com/AuthMe/AuthMeReloaded/issues/1095#issuecomment-335036850 sometime.
